src/main/webapp/app/components/commentIcon/CommentIcon.tsx

```diff
@@ -0,0 +1,160 @@
+import React, { useEffect, useMemo, useRef, useState } from 'react';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faComment } from '@fortawesome/free-solid-svg-icons/faComment';
+import { Comment } from 'app/shared/model/firebase/firebase.model';
+import { Button, Col, Container, Input, InputGroup, ListGroup, ListGroupItem, Popover, PopoverBody, Row } from 'reactstrap';
+import { SizeProp } from '@fortawesome/fontawesome-svg-core';
+import { CLOSE_POPOVER_DURATION_MILLISECONDS } from 'app/config/constants/constants';
+import './comment-icon.scss';
+
+export interface ICommentIconProps {
+  size: SizeProp;
+  comments: Comment[];
+  onCreateComment: (content: string) => void;
+  onResolveComment: (index: number) => void;
+  onUnresolveComment: (index: number) => void;
+  onDeleteComment: (index: number) => void;
+}
+
+function CommentIcon(props: ICommentIconProps) {
+  let color: string;
+  if (props.comments.length === 0) {
+    color = 'black';
+  } else if (props.comments.some(comment => comment.resolved === 'false')) {
+    color = 'gold';
+  } else {
+    color = 'green';
+  }
+
+  const [showComments, setShowComments] = useState(false);
+  const [timeoutId, setTimeoutId] = useState(null);
+  const [commentInputValue, setCommentInputValue] = useState('');
+
+  function handleMouseEnter() {
+    clearTimeout(timeoutId);
+    setShowComments(true);
+  }
+
+  function handleMouseLeave() {
+    const id = setTimeout(() => {
+      setShowComments(false);
+    }, CLOSE_POPOVER_DURATION_MILLISECONDS);
+
+    setTimeoutId(id);
+  }
+
+  return (
+    <div style={{ display: 'inline-block' }} onMouseEnter={handleMouseEnter} onMouseLeave={handleMouseLeave}>
+      <FontAwesomeIcon id="commentIcon" icon={faComment} color={color} />
+      <Popover target="commentIcon" isOpen={showComments}>
```

Comment:
Did you check out the rc-tooltip?

Reply:
Have not yet, mostly just wanted to get the functionality working, but should be an easy replace if we decide to go with tooltip